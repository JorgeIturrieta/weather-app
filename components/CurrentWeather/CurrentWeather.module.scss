@import 'styles/variables.scss';
.title {
  margin: 0.5rem 0;
}
.weatherHour {
  font-weight: bold;
}
.container {
  margin-top: 1rem;
  padding: 1rem;
  display: grid;
  border-radius: 4px;
  border: 1px solid tomato;
  -webkit-box-shadow: $shadow-card;
  box-shadow: $shadow-card;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  gap: 2rem;
  align-items: center;

  .description {
    gap: 1rem;
    display: flex;
    align-items: center;
  }
  .forecast {
    display: flex;
    gap: 1rem;
    flex-direction: column;
  }
  .temperature {
    font-size: 3rem;
  }
  .weatherDescription {
    color: gray;
    font-weight: 700;
    text-transform: uppercase;
  }

  .flex {
    gap: 0.5rem;
    display: flex;
    flex-direction: column;
  }
}
