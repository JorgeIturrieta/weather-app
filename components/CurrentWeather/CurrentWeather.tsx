import { Weather } from '@/types/weatherApi/apiWeahter.interface'
import getHumanReadableCountry from 'lib/getHumanReadableCountry'
import getHumanReadableTime from 'lib/getHumanReadableTime'
import Image from 'next/image'
import React, { FunctionComponent } from 'react'
import styles from './CurrentWeather.module.scss'
type CurrentWeatherProps = {
  data: Weather
}
export const CurrentWeather: FunctionComponent<CurrentWeatherProps> = ({ data }) => {
  return (
    <div>
      <h2>Pronostico para hoy</h2>
      <h3>
        {getHumanReadableTime(data.dt, data.timezone, {
          day: 'numeric',
          timeZone: 'UTC',
          weekday: 'long',
          month: 'long',
        })}{' '}
        en {data.name} , {getHumanReadableCountry(data.sys.country)}
      </h3>
      <div className={styles.container}>
        <div>
          <div className={styles.description}>
            <span className={styles.temperature}>{data.main.temp}°</span>
            <div>
              <Image
                src={`https://openweathermap.org//img/w/${data.weather[0].icon}.png`}
                alt={data.weather[0].description}
                width={80}
                height={80}
              />
            </div>
            <span className={styles.weatherDescription}>{data.weather[0].description}</span>
          </div>
          <span>
            {data.main.temp_min}° / {data.main.temp_max}°{' '}
          </span>
        </div>

        <div className={styles.forecast}>
          <span>Sensación termica: {data.main.feels_like}°</span>
          <span>
            {' '}
            Amanecer:{' '}
            {getHumanReadableTime(data.sys.sunrise, data.timezone, {
              hour: 'numeric',
              minute: 'numeric',
              timeZone: 'UTC',
            })}
            hs
          </span>
          <span>
            Puesta de Sol:{' '}
            {getHumanReadableTime(data.sys.sunset, data.timezone, {
              hour: 'numeric',
              minute: 'numeric',
              timeZone: 'UTC',
            })}
            hs
          </span>
        </div>
      </div>
    </div>
  )
}
