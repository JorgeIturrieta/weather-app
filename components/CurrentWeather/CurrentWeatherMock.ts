import { WeatherAdapted } from '../../types/weather.interface'
export const weatherMock: WeatherAdapted = {
  temp: 15.06,
  day: 'mi√©rcoles, 8 de junio',
  hour: '11:48',
  icon: '03d',
  description: 'nubes dispersas',
  temp_max: 15.06,
  temp_min: 15.06,
  feels_like: 13.59,
  sunrise: '8:29',
  sunset: '18:37',
  city: 'Villa Krause',
  country: 'Argentina',
  timezoneOffset: -10800,
}
