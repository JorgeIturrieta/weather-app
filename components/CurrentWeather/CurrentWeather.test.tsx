import { render, screen } from '@testing-library/react'
import { CurrentWeather } from './CurrentWeather'
import { weatherMock } from './CurrentWeatherMock'
describe('<CurrentWeather />', () => {
  it('Render CurrentWeather', async () => {
    render(<CurrentWeather data={weatherMock} />)

    const mainHeading = screen.getByRole('heading', { name: /pronóstico para hoy/i })
    const day = screen.getByRole('heading', {
      name: /miércoles, 8 de junio en villa krause, argentina/i,
    })
    const hour = screen.getByText('11:48 hs')
    const temperature = screen.getByText(/sensación termica: 13\.59°/i)
    expect(mainHeading).toBeInTheDocument()
    expect(day).toBeInTheDocument()
    expect(hour).toBeVisible()
    expect(temperature).toBeInTheDocument()
  })

  it('renders ForeCastWeather unchanged', () => {
    const { container } = render(<CurrentWeather data={weatherMock} />)
    expect(container).toMatchSnapshot()
  })
})
