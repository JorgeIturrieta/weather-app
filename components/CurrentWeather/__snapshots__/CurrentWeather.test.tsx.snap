// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentWeather /> renders ForeCastWeather unchanged 1`] = `
<div>
  <div>
    <h2>
      Pronóstico para hoy
    </h2>
    <h3
      class="title"
    >
      miércoles, 8 de junio
       en 
      Villa Krause
      , 
      Argentina
    </h3>
    <span>
      Hora local:
      <span
        class="weatherHour"
      >
         
        11:48
         hs
      </span>
    </span>
    <div
      class="container"
    >
      <div>
        <div
          class="description"
        >
          <div
            class="flex"
          >
            <span
              class="temperature"
            >
              15.06
              °
            </span>
            <span>
              15.06
              ° / 
              15.06
              °
               
            </span>
          </div>
          <div>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="nubes dispersas"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <span
            class="weatherDescription"
          >
            nubes dispersas
          </span>
        </div>
      </div>
      <div
        class="forecast"
      >
        <span>
          Sensación termica: 
          13.59
          °
        </span>
        <span>
           
          Amanecer: 
          8:29
          hs
        </span>
        <span>
          Puesta de Sol: 
          18:37
          hs
        </span>
      </div>
    </div>
  </div>
</div>
`;
